src/main/java/io/aiven/kafka/connect/s3/AwsCredentialProviderFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.s3;
+
+import io.aiven.kafka.connect.s3.config.AwsStsEndpointConfig;
+import io.aiven.kafka.connect.s3.config.AwsStsRole;
+import io.aiven.kafka.connect.s3.config.S3SinkConfig;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+
+public class AwsCredentialProviderFactory {
+    public AWSCredentialsProvider getProvider(final S3SinkConfig config) {
+        if (config.hasAwsStsRole()) {
+            return getStsProvider(config);
+        }
+        return getStaticProvider(config);
+    }
+
+    private AWSCredentialsProvider getStsProvider(final S3SinkConfig config) {
+        final AwsStsRole awsstsRole = config.getStsRole();
+        final AWSSecurityTokenService sts = securityTokenService(config);
+        return new STSAssumeRoleSessionCredentialsProvider.Builder(awsstsRole.arn, awsstsRole.sessionName)
+                .withStsClient(sts)
+                .withExternalId(awsstsRole.externalId)
+                .withRoleSessionDurationSeconds(awsstsRole.sessionDurationSeconds)
+                .build();
+    }
+
+    private AWSSecurityTokenService securityTokenService(final S3SinkConfig config) {
+        if (config.hasStsEndpointConfig()) {
+            final AwsStsEndpointConfig endpointConfig = config.getStsEndpointConfig();
+            final AwsClientBuilder.EndpointConfiguration stsConfig =
+                    new AwsClientBuilder.EndpointConfiguration(endpointConfig.serviceEndpoint,
+                                                               endpointConfig.signingRegion);
+            final AWSSecurityTokenServiceClientBuilder stsBuilder =
+                    AWSSecurityTokenServiceClientBuilder.standard();
+            stsBuilder.setEndpointConfiguration(stsConfig);
+            return stsBuilder.build();
+        }
+        return AWSSecurityTokenServiceClientBuilder.defaultClient();
+    }
+
+    private AWSCredentialsProvider getStaticProvider(final S3SinkConfig config) {
```

Comment:
A bit confused with naming, I think  `getBasicAWSCredentials`  would be better

Reply:
Maybe `getBasicAWSCredentialsProvider`?